### PR TITLE
doc(v2 data engine): add node disk (nvme, virtio and aio) support

### PR DESCRIPTION
**Pull request by derekbit**
#### Which issue(s) this PR fixes:

Issue Longhorn/longhorn#7672

### CODE REVIEW

File: content/docs/1.7.0/v2-data-engine/features/node-disk-support.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+- VirtIO Disk: managed by spdk_tgt using the virtio bdev driver, and `node.disks[i].diskDriver` will be set to `virtio-blk`.
+- Other Disks: managed by spdk_tgt using the aio bdev driver, and `node.disks[i].diskDriver` will be set to `aio`.
+
+Alternatively, users can manually set `node.disks[i].diskDriver` to `aio` to force the use of the aio bdev driver.
+
+To support NVMe and VirtIO disks, you need to find the BDF (Bus, Device, Function) of the disk as a disk path that will be added to the Longhorn node. The following examples provide an introduction to configuring NVMe disks, VirtIO disks, and others.
+
+### Using NVMe Disks
+
+1. List the disks
```
Comment (DamiaSan) [inline content/docs/1.7.0/v2-data-engine/features/node-disk-support.md:40]: newline needed

File: content/docs/1.7.0/v2-data-engine/features/node-disk-support.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+   Example output:
+   ```
+   lrwxrwxrwx  1 root root 0  Jul  30 12:20 loop0 -> ../devices/virtual/block/loop0
+   lrwxrwxrwx  1 root root 0  Jul  30 12:20 nvme0n1 -> ../devices/pci0000:00/0000:00:01.2/0000:02:00.0/nvme/nvme0/nvme0n1
+   lrwxrwxrwx  1 root root 0  Jul  30 12:20 nvme0n1 -> ../devices/pci0000:00/0000:00:01.2/0000:05:00.0/nvme/nvme1/nvme1n1
+   ```
+
+1. Get the BDF of the NVMe disk
+   Identify the BDF of the NVMe disk /dev/nvme1n1. From the example above, the BDF is 0000:05:00.0.
```
Comment (DamiaSan) [inline content/docs/1.7.0/v2-data-engine/features/node-disk-support.md:54]: newline needed

File: content/docs/1.7.0/v2-data-engine/features/node-disk-support.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+>    tags: []
+>  ```
+> In this case, the disk will be managed by the aio bdev driver, and the `node.disks[i].diskDriver` will be set to `aio`.
+
+### Using VirtIO Disks
+
+The steps are similar to NVMe disks.
+
+1. List the disks
+   First, identify the VirtIO disks available on your system by running the following command:
```
Comment (DamiaSan) [inline content/docs/1.7.0/v2-data-engine/features/node-disk-support.md:90]: newline needed

File: content/docs/1.7.0/v2-data-engine/features/node-disk-support.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+   lrwxrwxrwx  1 root root 0  Jul  30 12:20 loop0 -> ../devices/virtual/block/loop0
+   lrwxrwxrwx  1 root root 0 Feb 22 14:04 vda -> ../devices/pci0000:00/0000:00:02.3/0000:04:00.0/virtio2/block/vda
+   lrwxrwxrwx  1 root root 0 Feb 22 14:24 vdb -> ../devices/pci0000:00/0000:00:02.6/0000:07:00.0/virtio5/block/vdb
+   ```
+
+1. Get the BDF of the VirtIO disk
+   Identify the BDF of the VirtIO disk /dev/vdb. From the example above, the BDF is 0000:07:00.0.
+
+1. Add the NVMe disk to `spec.disks` of `node.longhorn.io`
+   ```
```
Comment (DamiaSan) [inline content/docs/1.7.0/v2-data-engine/features/node-disk-support.md:106]: newline needed

File: content/docs/1.7.0/v2-data-engine/features/node-disk-support.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+      diskType: block
+      evictionRequested: false
+      path: /dev/loop12
+      storageReserved: 0
+      tags: []
+    ```
+
+2. Check node.status.diskStatus. The disk should be detected without errors, and the `node.disks[i].diskDriver` will be set to aio.
+
+
```
Comment (DamiaSan) [inline content/docs/1.7.0/v2-data-engine/features/node-disk-support.md:153]: I should add a section where we say that, using nvme or virtio diskDriver, the disk will be detached by kernel driver and so the block device `/dev/nvmeXnY` or `/dev/vdbX` will disappear. WDYT?

Comment (derekbit) [reply content/docs/1.7.0/v2-data-engine/features/node-disk-support.md:153]: Sounds good. I will add it.

File: content/docs/1.7.0/v2-data-engine/features/node-disk-support.md (hunk 2 above)
Comment (roger-ryao) [inline content/docs/1.7.0/v2-data-engine/features/node-disk-support.md:54]:
```suggestion
   Identify the BDF of the NVMe disk `/dev/nvme1n1`. From the example above, the BDF is `0000:05:00.0`.
```

**Review by roger-ryao — CHANGES_REQUESTED** [review]

File: content/docs/1.7.0/v2-data-engine/features/node-disk-support.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+      allowScheduling: true
+      diskDriver: aio
+      diskType: block
+      evictionRequested: false
+      path: /dev/loop12
+      storageReserved: 0
+      tags: []
+    ```
+
+2. Check node.status.diskStatus. The disk should be detected without errors, and the `node.disks[i].diskDriver` will be set to aio.
```
Comment (roger-ryao) [inline content/docs/1.7.0/v2-data-engine/features/node-disk-support.md:151]:
```suggestion
2. Check `node.status.diskStatus`. The disk should be detected without errors, and the `node.disks[i].diskDriver` will be set to `aio`.
```

File: content/docs/1.7.0/v2-data-engine/features/node-disk-support.md [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+
+   Example output:
+   ```
+   lrwxrwxrwx  1 root root 0  Jul  30 12:20 loop0 -> ../devices/virtual/block/loop0
+   lrwxrwxrwx  1 root root 0 Feb 22 14:04 vda -> ../devices/pci0000:00/0000:00:02.3/0000:04:00.0/virtio2/block/vda
+   lrwxrwxrwx  1 root root 0 Feb 22 14:24 vdb -> ../devices/pci0000:00/0000:00:02.6/0000:07:00.0/virtio5/block/vdb
+   ```
+
+1. Get the BDF of the VirtIO disk
+   Identify the BDF of the VirtIO disk /dev/vdb. From the example above, the BDF is 0000:07:00.0.
```
Comment (roger-ryao) [inline content/docs/1.7.0/v2-data-engine/features/node-disk-support.md:103]:
```suggestion
   Identify the BDF of the VirtIO disk `/dev/vdb`. From the example above, the BDF is `0000:07:00.0`.
```

File: content/docs/1.7.0/v2-data-engine/features/node-disk-support.md [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+     allowScheduling: true
+     diskType: block
+     diskDriver: auto
+     evictionRequested: false
+     path: 0000:07:00.0
+     storageReserved: 0
+     tags: []
+   ```
+
+1. Check the `status.diskStatus`. The disk should be detected without errors, and the diskDriver should be set to `virtio-blk`.
```
Comment (roger-ryao) [inline content/docs/1.7.0/v2-data-engine/features/node-disk-support.md:117]:
```suggestion
1. Check the `status.diskStatus`. The disk should be detected without errors, and the `diskDriver` should be set to `virtio-blk`.
```

**Review by roger-ryao — CHANGES_REQUESTED** [review]

File: content/docs/1.7.0/v2-data-engine/features/node-disk-support.md (hunk 5 above)
Comment (derekbit) [reply content/docs/1.7.0/v2-data-engine/features/node-disk-support.md:168]: Added. Thank you.

**Pull request merged**